cpp/inline/mcts/Node.inl

```diff
@@ -246,14 +248,33 @@ void Node<Game>::update_stats(MutexProtectedFunc func) {
   func();
   lock.unlock();
 
-  core::seat_index_t cp = stable_data().current_player;
-
   ValueArray Q_sum;
   ValueArray Q_sq_sum;
   Q_sum.setZero();
   Q_sq_sum.setZero();
   int N = 0;
 
+  if (stable_data_.is_chance_node) {
+    for (int i = 0; i < stable_data_.num_valid_actions; i++) {
+      const edge_t* edge = get_edge(i);
+      const Node* child = get_child(edge);
+
+      if (!child) {
+        break;
+      }
+      const auto& child_stats = child->stats();
+      Q_sum += child_stats.Q * edge->chance_prob;
+      Q_sq_sum += child_stats.Q_sq * edge->chance_prob;
+      N++;
+    }
+    if (N == stable_data_.num_valid_actions) {
+      stats_.Q = Q_sum;
+      stats_.Q_sq = Q_sq_sum;
+    }
+
```

Comment:
Although the provably-winning/losing stuff is slated to be replaced (in my longstanding `Q-bounds` branch...), it's still used in `main`, and so needs to be properly maintained.

At a chance node, if every child is provably winning for player 1, then the parent is also provably winning for player 1. You are missing this computation.

The best way to fix this is to move the chance-branch inside the existing-for-loop, around the spots where `Q_sum` is updated.

Reply:
It occurred to me though that provably _losing_ is not calculated in the same way at chance nodes. It's rather done in the same way as provably winning (i.e., it's only provably losing if every child is expanded and every child is provably losing). So, there is an argument for keeping the branch you have, but just adding specialized calculations for just the provably stuff.